plugin/evm/gossiper.go

```diff
@@ -185,15 +186,51 @@ func (n *pushGossiper) queueRegossipTxs() types.Transactions {
 		)
 		return nil
 	}
-	localQueued := n.queueExecutableTxs(state, tip.BaseFee(), localTxs, n.config.TxRegossipMaxSize)
+	rgFrequency := n.config.RegossipFrequency
+	rgMaxTxs := n.config.RegossipMaxTxs
+	rgTxsPerAddr := n.config.RegossipTxsPerAddress
+	localQueued := n.queueExecutableTxs(state, tip.BaseFee(), localTxs, rgFrequency, rgMaxTxs, rgTxsPerAddr)
 	localCount := len(localQueued)
-	if localCount >= n.config.TxRegossipMaxSize {
+	if localCount >= rgMaxTxs {
 		return localQueued
 	}
-	remoteQueued := n.queueExecutableTxs(state, tip.BaseFee(), remoteTxs, n.config.TxRegossipMaxSize-localCount)
+	remoteQueued := n.queueExecutableTxs(state, tip.BaseFee(), remoteTxs, rgFrequency, rgMaxTxs-localCount, rgTxsPerAddr)
 	return append(localQueued, remoteQueued...)
 }
 
+// queueRegossipTxs finds the best priority transactions in the mempool and adds up to
+// [PriorityRegossipMaxTxs] of them to [txsToGossip].
+func (n *pushGossiper) queuePriorityRegossipTxs() types.Transactions {
+	// Fetch all pending transactions
+	pending := n.txPool.Pending(true)
```

Comment:
Would be something like this in mempool useful here to not actually traverse all other accounts (in both for effective tip filtering in mempool and here to prune other addresses)?
```
func (pool *TxPool) PendingFrom(from common.Address, enforceTips bool) types.Transactions {
...
}
```

Reply:
Ya that's a good idea, I'll add that change in.